From 57c9e00c28c667613397a9010f3a781df7287269 Mon Sep 17 00:00:00 2001
From: Tsz-Wo Nicholas Sze <szetszwo@hortonworks.com>
Date: Thu, 4 Aug 2016 09:45:40 -0700
Subject: [PATCH 1694/2848] HDFS-10716. In Balancer, the target task should be
 removed when its size < 0.  Contributed by Yiqun
 Lin

(cherry picked from commit 5004b0571e65f16b3913d57abdb63a0b7d8d09fe)

Change-Id: I4eea5f0d0a3b689d5d7377937f3b345f8af6dfbe
---
 .../hadoop/hdfs/server/balancer/Dispatcher.java    |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/balancer/Dispatcher.java b/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/balancer/Dispatcher.java
index ffdb9e1..eb4fb92 100644
--- a/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/balancer/Dispatcher.java
+++ b/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/balancer/Dispatcher.java
@@ -682,7 +682,7 @@ private PendingMove chooseNextMove() {
             long blockSize = pendingBlock.block.getNumBytes();
             incScheduledSize(-blockSize);
             task.size -= blockSize;
-            if (task.size == 0) {
+            if (task.size <= 0) {
               i.remove();
             }
             return pendingBlock;
-- 
1.7.9.5

